### ISSUE TITLE
Templated SMLNJ

### ISSUE THREAD

**Issue by isuruf** [spdx/license-list-XML#1441]
f2c has the same text with the copyright holder name different and a small text change
See http://www.netlib.org/f2c/src/Notice

It's a similar issue to https://github.com/spdx/license-list-XML/issues/994

Here's a diff of the two ignoring whitespace
```diff
@@ -1,18 +1,19 @@
 Permission to use, copy, modify, and distribute this software
 and its documentation for any purpose and without fee is hereby
 granted, provided that the above copyright notice appear in all
-copies and that both the copyright notice and this
+copies and that both that the copyright notice and this
 permission notice and warranty disclaimer appear in supporting
-documentation, and that the name of Lucent Technologies, Bell
-Labs or any Lucent entity not be used in
+documentation, and that the names of AT&T, Bell Laboratories,
+Lucent or Bellcore or any of their entities not be used in
 advertising or publicity pertaining to distribution of the
 software without specific, written prior permission.
 
-Lucent disclaims all warranties with regard to
+AT&T, Lucent and Bellcore disclaim all warranties with regard to
 this software, including all implied warranties of
-merchantability and fitness. In no event shall Lucent
-be liable for any special, indirect or consequential
+merchantability and fitness.  In no event shall AT&T, Lucent or
+Bellcore be liable for any special, indirect or consequential
 damages or any damages whatsoever resulting from loss of use,
 data or profits, whether in an action of contract, negligence or
 other tortious action, arising out of or in connection with the
 use or performance of this software.
 ```

**Comment by jakirkham** [comment]
IANAL

With the first diff line there is only an extra "that" (unless I'm missing something). It reads a bit funny having that there. Almost wonder if that was a typo.

The other differences look just like name substitutions. Though there are different kinds of references to "entities", which IDK if there is some nuance to those.

**Comment by jlovejoy** [comment]
Thanks @jakirkham - this is good material for some potential additions to our matching markup - we'll have a closer look!

**Comment by jakirkham** [comment]
Isuru deserves all the credit for spotting this connection :)

**Comment by johnmhoran** [comment]
Thanks @jlovejoy , @jakirkham and especially @isuruf -- this is a good catch.  

@jlovejoy -- I've taken a look at the similar `BSD-3-Clause-LBNL` issue and PR as well as @swinslow 's 4/5/20 commit https://github.com/swinslow/license-list-XML/commit/3f8f53f94e95577e2dd3615a6f91a79d0b0f5937.  Based on Steve's commit, I think I have a decent (but no doubt incomplete ;-) sense of the syntax to be used for the copyright template.  

I'm a bit rusty but if I understand/recall the process correctly, all I need to do is update my fork of the repo, checkout a new branch for issue #1441, add the templating language to [SMLNJ.xml](https://github.com/spdx/license-list-XML/blob/master/src/SMLNJ.xml), commit, push and open a PR.  Does that sound correct?  And since we're working with an existing license, no need to modify the .txt test file, right?

I'm not sure that I'll have time to tackle this before the Thursday call, but if I can, would you like me to give it a go, or wait for discussion on the call?

**Comment by johnmhoran** [comment]
@jlovejoy I've added template markup to the SMLNJ license `.xml` file, committed, pushed, and opened a PR.  Not 100% certain I applied the template syntax correctly, and in any event your comments and corrections are most welcome.  ;-)

1 of 2 tests failed, but I cannot decipher the failure -- something about duplicate licenses and something about `make: *** [Makefile:29: validate-canonical-match] Error 64`.  Is this a failure regarding the text, or a problem with our build/test process, or perhaps something else?

The PR does not show up here; for the record, it's https://github.com/spdx/license-list-XML/pull/1446.

As @jakirkham noted above, the `f2c` license has an extra `that` -- `. . . and that both that the copyright notice and this
permission notice . . .` -- which is clearly an error/typo.  I have not addressed the error in my work but its presence means the `f2c` text is not a 100% match to the newly-templated text -- how do we address that?

**Comment by swinslow** [comment]
Hi @johnmhoran -- so from a first quick look, I think this is related to the fact that there were originally two identical license texts with different identifiers added for this license. This predates me, but take a look at [this mailing list post from 2014](https://lists.spdx.org/g/Spdx-legal/message/979).

* [SMLNJ](https://spdx.org/licenses/SMLNJ.html) -- currently on the list
* [StandardML-NJ](https://spdx.org/licenses/StandardML-NJ.html) -- on the list but deprecated

Part of the CI system checks run a comparison between each license's text and all other licenses on the list, to confirm whether there are duplicates. It looks like that's where this is tripping up: [see here](https://github.com/spdx/license-list-XML/runs/6014166952?check_suite_focus=true#step:6:41)

Because there are expected to be some duplicated (e.g., each of the different ID's for GPL-3.0, etc.) there is a file in the repo called [expected-warnings](https://github.com/spdx/license-list-XML/blob/master/expected-warnings).

If you add that text: `Duplicates licenses: SMLNJ, StandardML-NJ` to the expected-warnings file, then that should cause this error to go away.

I haven't yet reviewed the rest of the PR and your comments above, but I think that's what's going on with this particular error  :)

**Comment by johnmhoran** [comment]
Thanks @swinslow.  I saw that error yesterday and mentioned it in my comment above but the meaning was elusive.  

I just added the text to the `expected-warnings` file, committed and pushed.  Never heard of this error or process -- is this documented anywhere?  Do we have a process that requires such a warning to be added as soon as a duplicate/deprecation is created?

**Comment by swinslow** [comment]
Fixed in #1446, thank you @johnmhoran!

@goneall has now added some details about the expected-warnings file in the `DOCS/new-license-workflow.md` documentation file, in #1447 -- thanks for suggesting that.
